[Bug 16407] Allow array split by empty delimiters

peter-b opened the pull request:
Modify `split` syntax to treat empty delimiters the same as delimiters that do not exist in the source string.  This is to harmonise its behaviour with that of the `contains` operator.

runrevmark commented:
The docs for split (and variants) should be updated to reflect the behavior of an empty delimiter.

peter-b committed:
[Bug 16407] Splitting by empty should be an error.

Modify the engine to make attempting to split a string with an empty
delimiter into an error.  This is to fix a regression relative to the
6.7 behaviour, and to prevent an infinite loop in `MCStringSplit`.

peter-b committed:
libfoundation: Fix Unicode string split with missing key delimiter

When splitting Unicode strings, make the behaviour when the key
delimiter is not found consistent with the behaviour for native
strings.

peter-b committed:
[Bug 16407] Allow splitting strings by empty

Make the behaviour of `split` when empty delimiters are used
consistent with the behaviour of `contains` -- i.e. the empty string
is never found in a string.  This means that splitting with an empty
delimiter is the same as splitting with a delimiter that doesn't exist
ing the string.

peter-b committed:
[Bug 16407] com.livecode.typeconvert: Test splitting by empty

peter-b committed:
[Bug 16407] Move split by empty check into Exec methods

peter-b committed:
Merge branch 'bugfix-16407-7.1' into bugfix-16407

peter-b committed:
libfoundation: MCStringFind() should never locate empty

`MCStringFind()` should never succeed in searching for an empty
needle.

peter-b committed:
libfoundation: Make MCStringSplit() handle empty delimiters

Modify `MCStringSplit()` to treat empty delimiters the same as
delimiters that are never located in the source string.

runrevmark commented:
@livecode-vulcan review ok 8c0ef1c

livecode-vulcan commented:
💙 review by @runrevmark ok 8c0ef1c9579be8b4d84d1df2ac448621caab3410

livecode-vulcan commented:
😎 test success 8c0ef1c9579be8b4d84d1df2ac448621caab3410
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1158)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1169)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1155)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1161)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1173)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1175)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1177)
